doc/development.md

```diff
@@ -89,4 +89,205 @@
+Continuation lines stick to the same rules as above. Closing
+brace/bracket/parenthesis shall line up under the first character of the
+line start.
+
+```python
+# Correct
+# Vertically aligned and extra 4 spaces for condition continuation
+foo = [var for var in bar_list
+       if condition_1 and
+           condition_2]
+
+# Closing bracket starts under the first character
+foo = [
+    var for var in bar_list if condition_1
+]
+```
+
+```python
+# Wrong
+# for-statement broken in multiple lines
+foo = [var for var
+       in bar_list if condition_1 and
+       condition_2]
+```
+
+### Docstrings
+
+Always use the three double-quote `"""` format for docstrings 
+(per [PEP 257](https://peps.python.org/pep-0257/)).
+A docstring should be organized as multiple summary lines (one physical line
+not exceeding 79 characters) terminated by a period. First and last line
+of the docstring shall be blank.
+
+Functions and classes must have a docstring, unless:
+- Not externally visible and very short.
+- Pertains to a test module.
+
+The docstring should be imperative-style (e.g., `Verify configuration file.`)
+rather than descriptive-style (e.g., `Verifies configuration file.`).
+The docstring should describe the function's main purpose, its calling syntax,
+and/or semantics, but avoid implementation details.
+
+Certain aspects of a function should be documented in special sections,
+listed below.
+
+#### Args: (or Arguments:)
+
+List each parameter by name. The type of the parameter may optionally follow
+the name, enclosed in parentheses, and then the description of the parameter
+separated by a colon. If the type is optional or can take multiple types,
+that can be specified with comma-separated lists or use
+[type hinting](https://docs.python.org/3/library/typing.html) syntax.
+If the description of the parameter is too long to
+fit in 79 characters, following hanging lines are indented with 4 extra
+spaces.
+
+#### Returns: (or Yields: for generators)
+
+Start with the type of the return value, followed by a colon and its semantics.
+If the function only returns None, this section is not required. It may also
+be omitted if the docstring starts with Return or Yield
+(e.g., `Return a boolean indicating if the folder has the right permissions.`),
+and the opening sentence is sufficient to describe the return value.
+
+#### Raises:
+
+List all exceptions that are relevant to the interface, followed by a
+description. Use a similar exception name + (type) + colon + space and hanging
+indent style as described in `Args:`.
+
+```python
+def example_function(arg_1, arg_2=None):
+    """
+    This is an example docstring. Start with summary explanation of the
+    function main purpose. Can expand multiple lines.
+
+    Args:
+        arg_1 (_type_): Long description can be split into
+            multiple lines with extra indentation.
+        arg_2 (_type_, optional): _description_. Defaults to None.
```

Comment:
That the argument is optional with None as the default value is already clear from the function declaration.

Reply:
I agree, "optional" and "Defaults to None" are not needed.